### ISSUE TITLE
I can't connect to Azurite for local development

### ISSUE THREAD

**Issue by esdrasbeleza** [Azure/azure-sdk-for-go#19791]
### Bug Report

- SDK version: 1.1.4
- Package: `github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus`
- go version: 1.19.4

I'm trying to connect to Azurite for local development using the connection string provided in [its docs](https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite?tabs=visual-studio).

Then I run:
```go
	connectionString := "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;QueueEndpoint=http://127.0.0.1:10001/devstoreaccount1"
	client, err := azservicebus.NewClientFromConnectionString(connectionString, nil)
	if err != nil {
		fmt.Println(err)
		return nil, err
	}

```

and I get the following error:

```
error while creating Azure Service Bus client: key "Endpoint" must not be empty
```

I believe the cause might be [here](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/azservicebus/internal/conn/conn.go#L78-L80): the parsing fails if we don't have `Endpoint=` specified.

**Comment by richardpark-msft** [comment]
Hi @esdrasbeleza, Azurite is an Azure Storage emulator, so the Service Bus package (azservicebus) isn't what you want.

You probably want something more like the blobs package: [azblob](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/storage/azblob).

If I'm misunderstanding, let me know.

**Comment by esdrasbeleza** [comment]
Ooops. No, I don't need the blobs package, but I need the Azure Queue Storage. Because its name says "Azure" and  "Queue", and [the Azurite page](https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite?tabs=visual-studio#azure-sdks) points to the list of SDKs in Go where I found this one, I thought I could use this library. ☹️ I seems that I was wrong, haha.

As an extra detail, my use case is that I have an Azure Function that consumes from Azure Queue Storage, so my new Golang app needs to write to it. I use Azurite to develop this Function, and in the last year it worked great.

I found [this other SDK](https://github.com/Azure/azure-storage-queue-go) — package `github.com/Azure/azure-storage-queue-go` —, unlisted, that seems to be what I want. But it's not updated since 2019 and is not listed in the SDKs page, so I'm a bit afraid of using it 😅

**Comment by richardpark-msft** [comment]
Ah, that makes total sense. I can definitely see how that's a bit confusing at the moment.

We don't yet have a monorepo (this repo) version of queues yet, so the one you've listed (github.com/Azure/azure-storage-queue-go) is the right one to use, at this point in time.

Closing as I think we've found the solution and understand the path that got us here, which we should be able to make clearer as time goes on.
